telemetry/reporting.go

```diff
@@ -0,0 +1,81 @@
+package telemetry
+
+import (
+	"time"
+)
+
+const Name string = "n"
+const Type string = "t"
+const Identifier string = "i"
+const Amount string = "a"
+const When string = "w"
+
+func NewReporter(requestsLog *RequestsLog, reportInterval int, reportURL string, telemetryEnabled bool) *Reporter {
+	return &Reporter{
+		log:      requestsLog,
+		interval: reportInterval,
+		url:      reportURL,
+		enabled:  telemetryEnabled,
+	}
+}
+
+type Reporter struct {
+	log      *RequestsLog
+	interval int
+	url      string
+	enabled  bool
+}
+
+// Reports function calls in the last <provided interval> seconds in CBOR format to the provided url.
+// Contains a failsafe mechanism in the case the url is unreachable.
+func (r *Reporter) Start() {
+	if r.enabled {
+		time.Sleep(time.Duration(r.interval) * time.Second)
+		extractedLog := r.log.ExtractLoggedRequests()
+
+		r.transformToOutputFormat(extractedLog)
+	}
+}
+
+// Transform the logged function calls to a minimized output format to reduce network traffic
+func (r *Reporter) transformToOutputFormat(logs *map[string]*RequestLog) {
+	transformer := NewOutputTransformer()
+	minimizedLogs := transformer.MinimizeFormat(logs)
+	/*cborLogs := */ transformer.EncodeAsCBOR(minimizedLogs)
+	// TODO:
```

Comment:
@JeroenStravers1 quick question, will you have issues for todo's?

Reply:
[Yes](https://github.com/creativesoftwarefdn/weaviate/issues/736). I figured I could keep my pushes contained to #699, but I noticed that I'm losing some detail/traceability by doing this.